R/plotResource.R

```diff
@@ -0,0 +1,152 @@
+#' Plot functions regarding the resources. Compatible with non mizerMR objects
+
+#' Plot the mortality applied on the resource spectrum(s)
+#'
+#' @param object An object of class \linkS4class{MizerSim} or
+#'   \linkS4class{MizerParams}.
+#' @param proportion A boolean value that determines whether values should be
+#' displayed as proportions from 0 to 1 or with their actual values. Default is TRUE.
+#' @param return_data A boolean value that determines whether the formatted data
+#' used for the plot is returned instead of the plot itself. Default value is FALSE
+#' @param ... Other arguments (currently unused)
+#' @return A ggplot2 object, unless `return_data = TRUE`, in which case a data
+#'   frame with the four variables 'w', 'value', 'Predator', 'Resource' is returned.
+#' @export
+#' @family plotting functions
+#' @seealso [plotting_functions]
+#' @examples
+#' \donttest{
+#' plotResourcePred(NS_params)
+#'
+#' # Returning the data frame
+#' fr <- plotResourcePred(NS_params, return_data = TRUE)
+#' str(fr)
+#' }
+plotResourcePred <- function(object, proportion = TRUE, return_data = FALSE)
+{
+    if (is(object, "MizerSim")) {
+        params <- object@params
+        params <- setInitialValues(params, object)
+    } else if (is(object, "MizerParams")) {
+        params <- validParams(object)
+    }
+    SpIdx <- factor(params@species_params$species,
+                    levels = params@species_params$species)
+
+    #How many backgrounds are being used? Assuming using only mizerMR for now for additional backgrounds
+    if(!is.null(getComponent(params, "MR")))
```

Comment:
In this case, let's just call the standard `plotResource()` function, which is now in mizerExperimental. Let's avoid code duplication where possible.